### DISCUSSION TITLE
Sierra SCI Adlib patch bank support (3.PAT / PATCH.003)

### DISCUSSION THREAD

**Discussion by MusicallyInspired** [Wohlstand/OPL3BankEditor#203] [Ideas]
I've requested this some time in the past, but is there any chance of getting Sierra's SCI Adlib patch bank format supported? We know the specs and the file format layout. The format also never changed throughout SCI's development from the very first game to the very last game so it works with all Sierra games that support Adlib/Sound Blaster MIDI music as an output option. The format is laid out here: http://sciwiki.sierrahelp.com/index.php?title=Adlib_Patch_Resource_Format_Specification

It's a fairly simple format. The header is just `8900h` signifying that it's an SCI patch bank resource file. From there the bank can have either 48 or 96 sets of instrument data which comprise 28 bytes each. If there are 96 instruments, the two "sub-banks" are separated by the magic numbers `ABCDh`. And that's basically it. The instrument data itself is just the standard 28 bytes from Adlib's BNK instrument data format referenced here: https://github.com/Wohlstand/OPL3BankEditor/blob/master/Specifications/HMI%20BNK%20Format.txt
(the last 28 bytes, that is. sans the first two unused bytes).

Two points:
- There are **_no instrument names_**
- The patches are **_melodic only_**

SCI never supported rhythm banks/patches and to get around it they used some of the melodic instrument entries for percussion sounds instead (usually the first 3 or 4 patches in a bank were drum sounds like Kick, Snare, Hihat, and Cymbal/Crash sometimes. Kick was often doubled for Tom.

As an SCI composer and tinkerer, it'd be really nice to be able to transfer/create Adlib sounds to/from/for SCI games from within a single program. Right now to edit/sample patches from SCI games you need to use the BNK2PAT and PAT2BNK console programs by NewRisingSun - which convert SCI bank resources to/from Adlib's BNK format - and then open them in OPL3 Bank Editor as BNK files.

**Comment by Wohlstand** [comment]
To implement this I will need some sample files at least to let me research them.

**Comment by Wohlstand** [comment]
P.S.:
> http://sciwiki.sierrahelp.com/index.php?title=Adlib_Patch_Resource_Format_Specification

This link is unavalable without proxy (it showing me 403). I enabled the proxy to dodge this block.

**Comment by MusicallyInspired** [comment]
Here are a couple example patch banks from Police Quest 2 (48 patches) and Space Quest 4 (96 patches).
[sci adlib patch banks.zip](https://github.com/Wohlstand/OPL3BankEditor/files/12246513/sci.adlib.patch.banks.zip)

And here are those same two banks pre-converted to Adlib BNK format with NewRisingSun's command line tools (also included for reference).
[pq2_sq4 BNK files.zip](https://github.com/Wohlstand/OPL3BankEditor/files/12246583/pq2_sq4.BNK.files.zip)

As well, here is a text-formatted version of the content of that link.
[sci patch format.txt](https://github.com/Wohlstand/OPL3BankEditor/files/12246541/sci.patch.format.txt)
